File: status.py
class Status:
    def __init__(self):
        self.map = {
            True: {
                'status': 'available',
                'color': 'green'
            },
            'xa': {
                'status': 'away',
                'color': 'yellow'
            },
            'dnd': {
                'status': 'do_not_disturb',
                'color': 'red'
            },
            False: {
                'status': 'offline',
                'color': None
            }
        }
        self.current = None

    def determine_status(self, response):
        if response['presence']:
            presence = response['presence']
            try:
                self.current = self.map[presence['show']]
            except KeyError:
                self.current = self.map[presence['is_online']]

            try:
                self.current['message'] = self.map['status']
            except KeyError:
                self.current['message'] = None
        else:
            self.current = self.map[False]
        return self

    def get_availability(self):
        return self.current['status']

    def get_color(self):
        return self.current['color']

File: src/status.js
module.exports = {
  parse: function(user) {
    if (!user) {
      return this.UNKNOWN;
    }

    if (user.presence) {
      if (user.presence.show == "chat") {
        return this.AVAILABLE;
      } else if (user.presence.show == "xa") {
        return this.AWAY;
      } else if (user.presence.show == "dnd") {
        return this.DO_NOT_DISTURB;
      }
      throw new Error();
    }

    return this.OFFLINE;
  },
  AVAILABLE: {
    "name": "available",
    "color": {
      "red": 0,
      "green": 255,
      "blue": 0
    }
  },
  AWAY: {
    "name": "away",
    "color": {
      "red": 255,
      "green": 255,
      "blue": 0
    }
  },
  DO_NOT_DISTURB: {
    "name": "do not disturb",
    "color": {
      "red": 255,
      "green": 0,
      "blue": 0
    }
  },
  OFFLINE: {
    "name": "offline",
    "color": {
      "red": 0,
      "green": 0,
      "blue": 0
    }
  },
  UNKNOWN: {
    "name": "unknown",
    "color": {
      "red": 0,
      "green": 0,
      "blue": 0
    }
  }
};
